src/passes/bytesConverter.ts

```diff
@@ -35,20 +39,24 @@ import {
 
 export class BytesConverter extends ASTMapper {
   visitExpression(node: Expression, ast: AST): void {
-    const typeNode = replaceFixedBytesType(getNodeType(node, ast.compilerVersion));
+    const typeNode = replaceBytesType(getNodeType(node, ast.compilerVersion));
     node.typeString = generateExpressionTypeString(typeNode);
     this.commonVisit(node, ast);
   }
 
   visitVariableDeclaration(node: VariableDeclaration, ast: AST): void {
-    const typeNode = replaceFixedBytesType(getNodeType(node, ast.compilerVersion));
+    const typeNode = replaceBytesType(getNodeType(node, ast.compilerVersion));
     node.typeString = generateExpressionTypeString(typeNode);
     this.commonVisit(node, ast);
   }
 
   visitElementaryTypeName(node: ElementaryTypeName, ast: AST): void {
     const typeNode = typeNameToTypeNode(node);
-    const replacementTypeNode = replaceFixedBytesType(typeNode);
+    if (typeNode instanceof StringType || typeNode instanceof BytesType) {
+      replaceNode(node, createArrayTypeName(createUint8TypeName(ast), ast));
```

Comment:
This should be ast.replaceNode, I should have been clearer when you mentioned that